visualize-admin/visualization-tool
# feat: Preview of charts in different screen sizes

Closes #1710.

This PR adds an option to preview every layout (tab, dashboard and single URLs) in three modes:
- full available window width,
- tablet width (`md` breakpoint),
- mobile width (`sm` breakpoint).

## How to test
1. Go to this link.
2. Add a few more charts.
3. Proceed to layout options.
4. ✅ See that there's a new toggle menu in the right top corner with device icons.
5. ✅ Click through them to see that the chart area adjusts accordingly – you can toggle off the preview by clicking on a currently select preview mode, or by `>>` button in the left top corner.
6. Switch to dashboard / free canvas layout and modify it.
7. Preview different viewport sizes and exit the preview mode.
8. ✅  See that the original layout was persisted.

## Next steps
We should probably persist the changes made in the preview mode, so that users can not only modify largest viewport layout (`lg`), but rather also smaller ones. Currently they are just preview, without an option to be saved.
